packages/core/src/types.ts

```diff
@@ -473,3 +473,36 @@ export interface InternalOptions<TProviderType = ProviderType> {
   cookies: CookiesOptions
   callbackUrl: string
 }
+
+/**
+ * Util type that matches some strings literally, but allows any other string as well.
+ * @source https://github.com/microsoft/TypeScript/issues/29729#issuecomment-832522611
+ */
+export type LiteralUnion<T extends U, U = string> =
+  | T
+  | (U & Record<never, never>)
+
+export interface SignInOptions extends Record<string, unknown> {
+  /**
+   * Specify to which URL the user will be redirected after signing in. Defaults to the page URL the sign-in is initiated from.
+   *
+   * [Documentation](https://next-auth.js.org/getting-started/client#specifying-a-callbackurl)
+   */
+  callbackUrl?: string
```

Comment:
Let's rename this to `redirectTo` (on the public side for now, we need to accommodate for this in core later).

We've been meaning to do this as it causes confusion:

https://github.com/nextauthjs/next-auth/issues/5305
https://github.com/nextauthjs/next-auth/issues/5308